src/NUnitEngine/nunit.engine.tests/Runners/MasterTestRunnerTests.cs

```diff
@@ -73,21 +84,57 @@ public void CleanUp()
                 _services.ServiceManager.Dispose();
         }
 
+        [Test]
+        public void CheckInternalRunner()
+        {
+            var prop = typeof(MasterTestRunner).GetField("_engineRunner", BindingFlags.NonPublic | BindingFlags.Instance);
+            var runner = prop.GetValue(_runner);
+#if NET35
```

Comment:
In most other places we use `!NETCOREAPP1_1 && !NETCOREAPP2_0`. Even though it will have the same behaviour now, it could make a difference in the future.

Reply:
My usage reflected the fact that I actually was only using it for .NET 3.5, so that's what I did first. I later added the net core tests based on what I saw it doing... I'm glad to change it if that's the more standard way. Of course, any new platform could change all of this logic and possibly require a new runner type.

Reply:
Personally, I like the consistency of using `!NETCOREAPP1_1 && !NETCOREAPP2_0` as we only have a few places where we switch on `NET35`, but I don't know if it is standard. I can remember that @jnm2 wrote something about it somewhere, but I cannot find it in this or the nunit repository or on the wiki, but it can be that my Google ability is not good enough. Let's see what Chris - or another - says.

Reply:
It would be tempting to add a define like SUPPORTS_APP_DOMAIN for the 3.5 build and use that for the places where that's the deciding factor.

Reply:
I added the change because I couldn't interpret the results I was seeing, particularly because I'm not familiar with the .NET Core builds. It does probably belong elsewhere and could be moved.

Reply:
Sorry, I'm (trying to be) not actually dead!

> I can remember that @jnm2 wrote something about it somewhere

The bottom of https://github.com/nunit/nunit/blob/master/BUILDING.md#defined-constants?

My rule of thumb is to try to name the odd one out, the one that is the reason that the condition is necessary. It's ambiguous to me many times though. Is supporting appdomains unusual, or is lack of support unusual?

Also consider `#if !(NETCOREAPP1_1 || NETCOREAPP2_0)` over `#if !NETCOREAPP1_1 && !NETCOREAPP2_0`.

`SUPPORTS_APP_DOMAIN` is even better especially if the same `#if` condition would need to be synchronized across more than location in the code if a new target was added.